### ISSUE TITLE
bugs when use parsecfg with arc

### ISSUE THREAD

**Issue by ringabout** [nim-lang/Nim#14020]
use parsecfg with arc

### Example
fails in arc but works in refc
```nim
import parsecfg, streams


proc loadPrologueEnv*(filename: string) =
  var f = newStringStream(filename)

  if f != nil:
    var p: CfgParser
    open(p, f, filename)
    while true:
      var e = p.next
      case e.kind
      of cfgEof:
        break
      else:
        raise
    f.close()

loadPrologueEnv("")
```

### Current Output
```
Traceback (most recent call last)
D:\QQPCmgr\Desktop\iNim\test.nim(20) test
D:\QQPCmgr\Desktop\iNim\test.nim(18) loadPrologueEnv
C:\Users\blue\.choosenim\toolchains\nim-1.2.0\lib\pure\streams.nim(178) close
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
Error: execution of an external program failed: 'D:\QQPCmgr\Desktop\iNim\test.exe '
```

### Expected Output
Works.

### Additional Information

It works with default gc.

This two examples works fine in both default gc and arc.
```nim
import parsecfg, streams


proc loadPrologueEnv*(filename: string) =
  var f = newStringStream(filename)

  if f != nil:
    var p: CfgParser
    open(p, f, filename)
    while true:
      var e = p.next
      case e.kind
      of cfgEof:
        break
      else:
        discard
    f.close()

loadPrologueEnv("")
```

```nim
import parsecfg, streams


proc loadPrologueEnv*(filename: string) =
  var f = newStringStream(filename)

  if f != nil:
    var p: CfgParser
    open(p, f, filename)
    while true:
      var e = p.next
      case e.kind
      of cfgEof:
        break
      else:
        raise
    # f.close()

loadPrologueEnv("")
```

```
$ nim -v
Nim Compiler Version 1.2.0
```

**Comment by cooldome** [comment]
Command line `--sinkInference:off` helps as workaround

**Comment by ringabout** [comment]
fixed in 1.2.2 and devel.
